source/agora/consensus/protocol/Nominator.d

```diff
@@ -521,3 +521,271 @@ extern(D):
         });
     }
 }
+
+/// Adds hashing support to SCPStatement
+private struct SCPStatementHash
+{
+    // sanity check in case a new field gets added.
+    // todo: use .tupleof tricks for a more reliable field layout change check
+    static assert(SCPNomination.sizeof == 80);
+
+    /// instance pointer
+    private const SCPStatement* st;
+
+    /// Ctor
+    public this (const SCPStatement* st) @nogc nothrow @trusted
+    {
+        assert(st !is null);
+        this.st = st;
+    }
+
+    /***************************************************************************
+
+        Compute the hash for SCPStatement.
+        Note: trusted due to union access.
+
+        Params:
+            dg = Hashing function accumulator
+
+    ***************************************************************************/
+
+    public void computeHash (scope HashDg dg) const nothrow @trusted @nogc
+    {
+        hashPart(this.st.nodeID, dg);
+        hashPart(this.st.slotIndex, dg);
+        hashPart(this.st.pledges.type_, dg);
+
+        switch (this.st.pledges.type_)
```

Comment:
```suggestion
        final switch (this.st.pledges.type_)
```

Reply:
I was gonna say that final switch sucks because of https://issues.dlang.org/show_bug.cgi?id=11051

But then I saw that the issue had a PR.. https://github.com/dlang/dmd/pull/6095

lol